src/core/connection.c

```diff
@@ -1198,66 +1167,32 @@ QuicConnTimerCancel(
     _In_ QUIC_CONN_TIMER_TYPE Type
     )
 {
-    for (uint32_t i = 0;
-        i < ARRAYSIZE(Connection->Timers) &&
-            Connection->Timers[i].ExpirationTime != UINT64_MAX;
-        ++i) {
+    CXPLAT_DBG_ASSERT(Connection->EarliestExpirationTime <= Connection->ExpirationTimes[Type]);
 
+    if (Connection->EarliestExpirationTime == UINT64_MAX) {
```

Comment:
I'm wondering how necessary this check is... It might be simpler just to remove it.